ObjectMapper/Core/Mapper.swift

```diff
@@ -102,6 +102,13 @@ public final class Mapper<N: BaseMappable> {
 			}
 		}
 
+		if let klass = N.self as? ImmutableMappable.Type {
+			if let maybeObject = try? klass.init(map: map) as? N, var object = maybeObject {
```

Comment:
I tried this out, and while it does make sense, it also seems like there will be a lot of duplication. In additional to the `Mapper` methods, the `Array` and `Set` convenience methods would have to be duplicated as well (if I am correct).

I also found that this alternative solution only works when the compile-time type of the object was `ImmutableMappable` - in my particular use case I have a manager class that took in request objects whose only generic constraint was `BaseMappable`. So it seems like I won't be able to interchange `Mappable` and `ImmutableMappable` classes easily. Not that I really needed to anyways, since all my mappable objects are immutable now.